README.rst

```diff
@@ -27,102 +18,148 @@ Getting Started
+Make sure you have installed all of the following prerequisites on your development machine:
 
 - `Docker <https://www.docker.com/>`__
-
-For Testing on GPU
-^^^^^^^^^^^^^^^^^^
-
--  `Nvidia-Docker <https://github.com/NVIDIA/nvidia-docker>`__
+- For GPU testing: `nvidia-docker2 <https://github.com/NVIDIA/nvidia-docker>`__
 
 Recommended
 ^^^^^^^^^^^
 
--  A python environment management tool. (e.g.
-   `PyEnv <https://github.com/pyenv/pyenv>`__,
+-  A Python environment management tool (e.g. `PyEnv <https://github.com/pyenv/pyenv>`__,
    `VirtualEnv <https://virtualenv.pypa.io/en/stable/>`__)
 
-Building your image
--------------------
+Building Images
+---------------
 
-`Amazon SageMaker <https://aws.amazon.com/documentation/sagemaker/>`__
-utilizes Docker containers to run all training jobs & inference endpoints.
+The Dockerfiles in this repository are intended to be used for building Docker images to run training jobs and inference endpoints on `Amazon SageMaker <https://aws.amazon.com/documentation/sagemaker/>`__.
 
-The Docker images are built from the Dockerfiles specified in
-`Docker/ <https://github.com/aws/sagemaker-mxnet-containers/tree/master/docker>`__.
+The current master branch of this repository contains Dockerfiles and support code for MXNet versions 1.3.0 and higher.
+For MXNet versions 0.12.1-1.2.1, check out v1.0.0 of this repository.
 
-The Docker files are grouped based on MXNet version and separated
-based on Python version and processor type.
+For each supported MXNet version, Dockerfiles can be found for each processor type (i.e. CPU and GPU).
+For MXNet versions 0.12.1 and 1.0.0, there are separate Dockerfiles for each Python version as well.
 
-The Docker images, used to run training & inference jobs, are built from
-both corresponding "base" and "final" Dockerfiles.
+All images are tagged with <mxnet_version>-<processor>-<python_version> (e.g. 1.3.0-cpu-py3).
 
-Base Images
-~~~~~~~~~~~
+MXNet 1.1.0 and higher
+~~~~~~~~~~~~~~~~~~~~~~
 
-The "base" Dockerfile encompass the installation of the framework and all of the dependencies
-needed.
+For these MXNet versions, there is one set of Dockerfiles for each version.
+They install the SageMaker-specific support code found in this repository.
 
-Tagging scheme is based on <mxnet_version>-<processor>-<python_version>. (e.g. 0.12.1-cpu-py2)
+Before building these images, you need to have two files already saved locally.
+The first is a pip-installable binary of the MXNet library.
+This can be something you compile from source or `download from PyPI <https://pypi.org/project/mxnet/#files>`__.
 
-All "final" Dockerfiles build images using base images that use the tagging scheme
-above.
+The second is a pip-installable binary of this repository.
+To create the SageMaker MXNet Container Python package:
+
+::
+
+    # Create the binary
+    git clone https://github.com/aws/sagemaker-mxnet-container.git
+    cd sagemaker-mxnet-container
+    python setup.py sdist
```

Comment:
https://github.com/aws/sagemaker-pytorch-container/blob/master/README.rst#final-images

Looks like we have inconsistencies for how we build. In PyTorch we do bdist...

Reply:
interesting. bdist builds a binary specific to the platform ([source](https://stackoverflow.com/a/6292860/9074534)), so in this case, I think sdist would be better since my Mac (for example) will have a different platform from the container.